test/cli/test_login.rb

```diff
@@ -0,0 +1,272 @@
+# frozen_string_literal: true
+
+require_relative "test_helper"
+require "aruba"
+
+class CLILoginTest < Minitest::Test
+  include Aruba::Api
+  include Helpers
+
+  def setup
+    setup_aruba
+  end
+
+  def test_login
+    label = "test-login"
+    username = "user"
+    password = "pass"
+    output = <<~OUTPUT
+      Label: #{label}
+      \tUsername: #{username}
+      \tPassword: #{password}
+    OUTPUT
+
+    # create login:
+    run_command "kbsecret new login -x #{label}" do |cmd|
+      cmd.stdin.puts "#{username}:#{password}"
+      cmd.stdin.close
+      cmd.wait
+    end
+
+    # retrieve login:
+    run_command "kbsecret login #{label}" do |cmd|
+      cmd.wait
+      assert_equal output, cmd.output
+    end
+  ensure
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
+  end
+
+  def test_terse
+    label = "test-terse"
+    username = "user"
+    password = "pass"
+    output = "#{label}:#{username}:#{password}"
+
+    # create login:
+    run_command "kbsecret new login -x #{label}" do |cmd|
+      cmd.stdin.puts "#{username}:#{password}"
+      cmd.stdin.close
+      cmd.wait
+    end
+
+    # retrieve login tersely:
+    run_command "kbsecret login -x #{label}" do |cmd|
+      cmd.wait
+      assert_equal output, cmd.output.chomp
+    end
+  ensure
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
+  end
+
+  def test_terse_separator
+    label = "test-terse-separator"
+    username = "user"
+    password = "pass"
+    separator = "^"
+    output = [label, username, password].join separator
+
+    # create login:
+    run_command "kbsecret new login -x #{label}" do |cmd|
+      cmd.stdin.puts "#{username}:#{password}"
+      cmd.stdin.close
+      cmd.wait
+    end
+
+    # retrieve login with separator:
+    run_command "kbsecret login -i #{separator} -x #{label}" do |cmd|
+      cmd.wait
+      assert_equal output, cmd.output.chomp
+    end
+  ensure
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
+  end
+
+  def test_all
+    session = "test-session"
+    label = "test-session-login"
+    label1 = "test-session-login1"
+    username = "user"
+    password = "pass"
+   # NOTE: need regex since we can't be sure of the order in which the logins will be retreived
+    pattern = /Label: (#{label}|#{label1})\n\tUsername: #{username}\n\tPassword: #{password}\nLabel: (#{label1}|#{label})\n\tUsername: #{username}\n\tPassword: #{password}\n/
```

Comment:
Looking at this again, this sort of pattern will be difficult to update/maintain if the output changes. It might be better to take the output from the command and analyze each line individually instead (i.e., via a small state machine).

Reply:
maybe we can iterate on this one? i have a sizable backlog of additional tests to PR once PR #38 gets merged

Reply:
That's fine -- just mark it with a `NOTE` or `TODO`.